app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/twofactorlogin/TwoFactorLoginNavigation.kt

```diff
@@ -34,8 +34,11 @@ fun NavController.navigateToTwoFactorLogin(
     password: String?,
     navOptions: NavOptions? = null,
 ) {
+    val requiredRoute = "$TWO_FACTOR_LOGIN_PREFIX/$emailAddress"
+    // only add the password if it is not null.
+    val routeOptions = password?.let { "?$PASSWORD=${it.base64UrlEncode()}" } ?: ""
```

Comment:
Is there something inherently wrong with `?$PASSWORD=null`?

Reply:
the arg would then be `args[PASSWORD] == "null"` only in this case "null" gets encoded, then we are passing that string value to the "login( ..., ..., password = theMessIDescribeAbove)"

Reply:
I wonder if we are just missing this?

```kotlin
        arguments = listOf(
            navArgument(EMAIL_ADDRESS) { type = NavType.StringType },
            navArgument(PASSWORD) {
                type = NavType.StringType
                nullable = true
            },
        ),
```

Reply:
you're 100% correct

Reply:
I am going to remove the tests, because A: we are just using the compose navigation here so no need to test it and B: it breaks the null case : {